index.js

```diff
@@ -4,7 +4,7 @@ const { execSync } = require('child_process');
 
 const host = core.getInput('host');
 const port = core.getInput('port');
-const socketPath = core.getInput('socket-path');
+const socketPath = core.getInput('socket-path') || execSync('mktemp 2> /dev/null', {encoding: 'utf-8'});
```

Comment:
I think `mktemp` creates the file, and I think `ssh-agent` won't create a socket over a path that already exists.

If it does turn out that's the case, you can use `mktemp -d` instead and give `ssh-agent` a path inside of there.

Reply:
```
Run sourcetoad/ssh-socket-action@SIP-388-dist
Attempting to create /tmp/tmp.hDJm78G7Ya
...
# github.com:22 SSH-2.0-babeld-901a823a
# github.com:22 SSH-2.0-babeld-901a823a
# github.com:22 SSH-2.0-babeld-901a823a
# github.com:22 SSH-2.0-babeld-901a823a
# github.com:22 SSH-2.0-babeld-901a823a
Identity added: (stdin) (xxx-action)
```

Do you think its platform specific? It seemed to use my existing file that `mktemp` made without an issue.

Reply:
Possibly. On mac os I get
```
$ ssh-agent -a `mktemp`
unix_listener: cannot bind to path /var/folders/0v/hmrgxkt5183g9z7yy_4hvg1m0000gn/T/tmp.T7io84S: Address already in use
```

But if I use a path without an existing file (e.g. remove one char from the end of the tmp path), it works

I'm honestly surprised that it would be happy to use an existing file

Reply:
Moved to `-u`.

Reply:
I still think something might be up, here. I think it shouldn't have been working before. Also, why swallow an error from mktemp? Could we instead save in a variable and catch an error thrown?

Reply:
Now there is error handling and a self-testing dogfood ci pipeline.